pydantic_ai_slim/pydantic_ai/durable_exec/temporal/_agent.py

```diff
@@ -242,10 +242,21 @@ def temporal_activities(self) -> list[Callable[..., Any]]:
         return self._temporal_activities
 
     @contextmanager
-    def _temporal_overrides(self) -> Iterator[None]:
-        # We reset tools here as the temporalized function toolset is already in self._toolsets.
+    def _temporal_overrides(self, toolsets: Sequence[AbstractToolset[AgentDepsT]] | None = None) -> Iterator[None]:
+        in_workflow = workflow.in_workflow()
+
+        if toolsets:
+            if in_workflow and any(not isinstance(t, TemporalWrapperToolset) for t in toolsets):
```

Comment:
- `CombinedToolset`s and others that don't require temporal wrappers are fine, so we should look at the ones covered by `temporalize_toolset`
- We should look down the hierarchy of toolsets using `toolset.visit()`, if we find a temporal wrapper, we know we don't need to look at its children, but otherwise we error if we encountered an unwrapped toolset